sympy/ntheory/elliptic_curve.py

```diff
@@ -263,6 +280,313 @@ def rank(self):
+            p5 = exp(p5, p, fl)
+            p6 = short_com["y2_p2"]
+            if k % 2 == 0:
+                p19 = (p3*alpha*b2 - b3*p6 - y2*a3) % fl
+                p19 = p4*(p19*y2*p2 - p1) % fl
+                if tau % 2 == 0:
+                    p19 *= exp(y2, (3*p + 1) // 2, fl)
+                    p19 = (p19 - b3*y2*p2*p5) % fl
+                else:
+                    yq = exp(y2, (p + 1) // 2, fl)
+                    p19 = (p19 - b3*p2*p5*yq) % fl
+            else:
+                y4 = y2**2 % fl
+                p7 = b3*y4*p6 % fl
+                p19 = (p3*alpha*b2*y2 - p7- a3) % fl
+                p19 = (p4*(p19*p2 - y4*p1)) % fl
+                if tau % 2 == 0:
+                    p19 *= exp(y2, (3*p - 3) // 2, fl)
+                    p19 = (p19 - b3*p2*p5) % fl
+                else:
+                    yq = exp(y2, (p + 3) // 2, fl)
+                    p19 = (p19 - b3*p2*p5*yq) % fl
+            return p19
+
+        def case1(l):
+            from sympy.ntheory import is_quad_residue, sqrt_mod
+            if not is_quad_residue(p, l):
+                return 0
+            w = sqrt_mod(p, l)
+            fl = f(l)
+            p17 = eq17(l, w)
+            g = gcd(p17, fl)
+            if g == 1:
+                return 0
+            p18 = eq18(l, w)
+            g = gcd(p18, fl)
+            if g == 1:
+                t1 = -2*w % l
+            else:
+                t1 = 2*w % l
+            return t1
+
+        def tmod2(p):
+            xp = exp(poly(X, X, modulus=p), p, y2)
+            g = gcd(xp-poly(X, X, modulus=p) , y2)
+            if g == 1:
+                return 1
+            return 0
+        def _schoof():
+            from sympy.ntheory.modular import solve_congruence
+            t1 = tmod2(p)
+            t = {2 : t1}
+            for i in range(1, len(L)):
+                l = L[i]
+                fl= f(l)
+                short_com = {}
+                k = p % l
+                short_com["p1"] = f(k - 1)*f(k + 1) % fl
+                short_com["p2"] = exp(poly(X, X, modulus=p), p, fl)
+                short_com["p3"] = exp(short_com["p2"], p, fl)
+
+                p16 = eq16(l, short_com)
+                g = gcd(p16, fl)
+                if g != 1:
+                    t1 = case1(l)
+                    t[l] = t1
+                else:
+                    if k not in f2:
+                        f2[k] = f2(k)**2
+                    short_com["p5"] = f2[k] % fl
+                    short_com["y2_p2"] = exp(y2, (p**2 - 1) // 2, fl)
+                    alpha = alpha_cal(l, short_com)
+                    beta = beta_cal(l, short_com)
+                    short_com["p4"] = short_com["p3"] + short_com["p2"] + poly(X, X, modulus=p)
+
+                    short_com["p9"] = exp(y2, p, fl)
+                    short_com["alpha2"] = alpha**2 % fl
+                    short_com["beta2"] = beta**2 % fl
+                    short_com["p5p4"] = short_com["p4"]*short_com["p5"] % fl
```

Comment:
It think it could be cleaner to just use variables for these, and pass them through to the various functions as parameters.

Reply:
I will make the    changes.